app/(auth)/signup/page.tsx

```diff
@@ -0,0 +1,131 @@
+'use client';
+
+import { useState } from 'react';
+import PasswordComplexity from '@/app/utils/PasswordComplexity';
+import PasswordInput from '@/app/utils/PasswordInput';
+import { AuthProvider, useAuth } from '../../utils/AuthProvider';
+
+export default function SignUpLayout() {
+  return (
+    <AuthProvider>
+      <SignUp />
+    </AuthProvider>
+  );
+}
+
+function SignUp() {
+  const { signUp } = useAuth();
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState<string | null>(null);
+  const [confirmPassword, setConfirmPassword] = useState<string | null>(null);
+  const [passwordError, setPasswordError] = useState('');
+  const [passwordComplexityError, setPasswordComplexityError] = useState<
+    string | null
+  >(null);
+  const [passwordComplexity, setPasswordComplexity] = useState<boolean>(false);
```

Comment:
would it be easier to change these variables to more descriptive names? lowkey got a bit confused haha, maybe `isPasswordComplexityMet` or something similar would help to differentiate it from `passwordComplexityError`

Reply:
good point! agreed about renaming